server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -256,6 +258,36 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         "search.concurrent_segment_search.enabled",
         false,
         Property.Dynamic,
+        Property.NodeScope,
+        Property.Deprecated
+    );
+
+    // Allow concurrent segment search for all requests
+    public static final String CONCURRENT_SEGMENT_SEARCH_MODE_ALL = "all";
+
+    // Disallow concurrent search for all requests
+    public static final String CONCURRENT_SEGMENT_SEARCH_MODE_NONE = "none";
+
+    // Make decision for concurrent search based on concurrent search deciders
+    public static final String CONCURRENT_SEGMENT_SEARCH_MODE_AUTO = "auto";
```

Comment:
Maybe we could use an `Enum` instead here

Reply:
Well these are just constants, I could create a `Enum` but I think thats just an overkill for this.; given that i also use this in static imports at other places.